src/main/java/org/broadinstitute/consent/http/service/ElectionService.java

```diff
@@ -94,17 +98,13 @@ public List<Election> describeClosedElectionsByType(String type, AuthUser authUs
             elections = dacService.filterElectionsByDAC(
                     electionDAO.findLastDataAccessElectionsWithFinalVoteByStatus(ElectionStatus.CLOSED.getValue()),
                     authUser);
-            List<String> referenceIds = elections.stream().map(Election::getReferenceId).collect(Collectors.toList());
-            List<DataAccessRequest> dataAccessRequests = dataAccessRequestService.
-                    getDataAccessRequestsByReferenceIds(referenceIds);
             elections.forEach(election -> {
-                Optional<DataAccessRequest> darOption = dataAccessRequests.stream().
-                        filter(d -> d.getReferenceId().equals(election.getReferenceId())).
-                        findFirst();
-                darOption.ifPresent(dar -> {
-                    election.setDisplayId(dar.getData().getDarCode());
+                DarCollection collection = darCollectionDAO.findDARCollectionByReferenceId(election.getReferenceId());
```

Comment:
This is a DAO call in a loop and will be pretty expensive. Instead, we should query for all of the collections before iterating over the elections and then pull the right collection from the list while inside the loop.